lib/scanner.js

```diff
@@ -1,24 +1,26 @@
-const { isWhitespace } = require('./type-checks')
+const { isWhitespace, isNewline } = require('./type-checks')
 const { CR, LF } = require('./codes')
 
 class Scanner {
-  constructor (text) {
-    this.pos = 0
+  constructor (text, pos) {
     this.text = text
+    this.pos = pos ? { ...pos } : { line: 1, column: 1, offset: 0 }
```

Comment:
`line` and `column` starts at `1`, `offset` starts at `0`. The `offset` is the original value of `this.pos` 😄